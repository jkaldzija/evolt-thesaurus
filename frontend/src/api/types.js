export type Word = {
    synonyms: Array<string>,
    value: string
}

export type WordListPage = {
    count: number,
    words: Array<string>
}
