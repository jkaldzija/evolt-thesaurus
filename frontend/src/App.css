.add-button {
    position: fixed;
    bottom: 48px;
    right: 48px;
}

.app-container {
    margin-top: 120px;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
    max-width: 960px;
    padding-left: 24px;
    padding-right: 24px;
}

.synonym-list-container {
    padding: 24px;
    margin-top: 36px;
}




.header-title{
    font-family: "Museo Sans Rounded 700" !important;
    font-weight: bold;
}

.primary{
    /*background-color: #4FADAB !important;*/
    background-color: #3cbd3f !important;
    /*background-color: #F0544F !important;*/
}

.black{
    color: #303030 !important;
}




.va-text{
    display: table-cell;
    vertical-align: middle;
}

*:focus {
    outline: none;
}

textarea:focus, input:focus{
    outline: none !important;
    -webkit-box-shadow: none !important;
    -moz-box-shadow: none !important;
    box-shadow: none !important;
    border-color: unset !important;
}

.card{
    border-radius:10px !important;
    padding: 35px 40px 40px 40px;
}

.modal-content{
    border-radius: 10px;
    padding: 15px 20px 20px 20px;
}

.typography-bold{
    font-family: "Museo Sans Rounded 500" !important;
}

.typography-extra-bold{
    font-family: "Museo Sans Rounded 700" !important;
}

body{
    background-color: rgb(233,233,233);
    font-family: "Museo Sans Rounded 300";
    padding-bottom: 110px;
}

.toast{
    border-radius: 10px;
}
