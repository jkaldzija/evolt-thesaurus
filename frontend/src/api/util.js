import Axios, {AxiosInstance} from 'axios'
import {API_URL} from "../config";

export const apiClient: AxiosInstance = Axios.create({
    baseURL: API_URL
});
