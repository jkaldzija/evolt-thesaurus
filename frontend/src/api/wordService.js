import {apiClient} from "./util";
import {AxiosPromise, AxiosResponse} from "axios";
import type {Word, WordListPage} from "./types";

// Define API paths in constants
export const WORD_PATH = "word";
export const WORD_LIST_PATH = "word/list";


export default {

    //List all synonyms for a word
    getSynonymsForWord: (value: string): AxiosPromise<AxiosResponse<Word>> => apiClient.get(WORD_PATH, {
        params: {
            value
        }
    }),

    // Create/Update a word with its synonyms
    createWord: (word: Word): AxiosPromise<AxiosResponse<string>> => apiClient.post(WORD_PATH, word),

    // List the words in the dictionary with pagination
    listWordsPaginated: (offset: number, limit: number): AxiosPromise<AxiosResponse<WordListPage>> => apiClient.get(WORD_LIST_PATH, {
        params: {
            offset,
            limit
        }
    })
}
