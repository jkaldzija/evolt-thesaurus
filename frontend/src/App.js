import React, {Component, Fragment} from 'react';
import 'bootstrap/dist/css/bootstrap.css';
import './App.css';
import './fonts.css';
import {Col, Container, Row} from "reactstrap";
import {ToastContainer} from "react-toastify";
import SearchBox from "./components/SearchBox/SearchBox";
import wordService from "./api/wordService";
import 'react-toastify/dist/ReactToastify.css';
import Header from "./components/Header/Header";
import {Route, Switch, withRouter} from "react-router";
import AddWordView from "./components/AddWord/AddWordView";
import {routes} from "./components/Utility/routing";
import Home from "./components/Views/Home";
import Dictionary from "./components/Views/Dictionary";


export const navigationRoutes = [
    {
        path: routes.home.path,
        component: Home
    },
    {
        path: routes.dictionary.path,
        component: Dictionary
    }
];

class App extends Component {
    state = {
        addDialogOpen: false,
        word: null
    };

    toggleAddDialog = () => this.setState({
        addDialogOpen: !this.state.addDialogOpen
    });

    onSubmit = async (word) => {
        const response = await wordService.getSynonymsForWord(word);
        this.setState({word: response.data});
    };

    render() {
        const {addDialogOpen, word} = this.state;
        return (
            <div>
                <ToastContainer/>
                <Header />
                <Container>
                    <Switch>
                        {navigationRoutes.map((routeItem, index) => {
                            const TagName = routeItem.component;
                            return <Route key={routeItem.path} exact path={routeItem.path} component={TagName} />
                        })}
                    </Switch>
                </Container>
            </div>
        );
    }
}

export default withRouter(App);
