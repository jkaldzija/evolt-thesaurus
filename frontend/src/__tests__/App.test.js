import React from 'react';
import {mount, shallow} from "enzyme";
import {apiClient} from "../api/util";
import App from "../App";
import MockAdapter from "axios-mock-adapter";
import {MemoryRouter, withRouter} from "react-router";

describe('Application', function () {
    it('should render correctly', function () {
        const component = shallow(
            <MemoryRouter>
                <App/>
            </MemoryRouter>

        ).dive().dive().dive().dive();
        component.setState({
            word: {
                value: "test",
                synonyms: [
                    "syn1",
                    "syn2",
                    "syn3"
                ]
            }
        });
        expect(component).toMatchSnapshot();
    });
    it('should open dialog on fab click', function () {
        const component = shallow(
            <MemoryRouter>
                <App/>
            </MemoryRouter>

        ).dive().dive().dive().dive();
        component.instance().toggleAddDialog();
        expect(component.state("addDialogOpen")).toBe(true);
        component.instance().toggleAddDialog();
        expect(component.state("addDialogOpen")).toBe(false);
    });
    it('should handle search', async function () {
        const wordResponse = {
            value: "test",
                synonyms: [
                "syn1",
                "syn2",
                "syn3"
            ]
        };
        const mock = new MockAdapter(apiClient);
        mock.onGet("word", {value: "test"}).reply(200, wordResponse);
        const component = shallow(
            <MemoryRouter>
                <App/>
            </MemoryRouter>

        ).dive().dive().dive().dive();
        await component.instance().onSubmit("test");
        expect(component.state("word")).toEqual(wordResponse);
    });
});
