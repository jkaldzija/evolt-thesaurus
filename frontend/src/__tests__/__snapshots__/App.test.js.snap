// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application should render correctly 1`] = `
<div>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <Header />
  <Container
    tag="div"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        key="/"
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        key="/dictionary"
        path="/dictionary"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </Container>
</div>
`;
